### PR TITLE
DDF-3930 Deprecating ConfigurationFileProxy and EmbeddedSolrFiles

### PR DESCRIPTION
**Pull request by austinsteffes**
#### What does this PR do?
Adds @Deprecates to ConfigurationFileProxy and EmbeddedSolrFiles to account for deprecation of their usage based on solr.data.dir.
#### Who is reviewing it? 
(please choose AT LEAST two reviewers that need to approve the PR before it can get merged)
@mcalcote 
@garrettfreibott 
@ahoffer 
#### Select relevant component teams: 
https://github.com/orgs/codice/teams
#### Ask 2 committers to review/merge the PR and tag them here. If you don't know who to ask, you can request reviews in https://groups.google.com/forum/#!forum/ddf-developers .
(please choose ONLY two committers from below, delete the rest)
@coyotesqrl 
#### How should this be tested? (List steps with links to updated documentation)
build ddf
#### What are the relevant tickets?
[DDF-3930](https://codice.atlassian.net/browse/DDF-3930)
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by austinsteffes** [comment]
build now

**Comment by paouelle** [comment]
❓ Why are we deprecating them instead of removing them? If they are no longer being used internally then let's cleanup since this is not part of the whitelist anyway

**Comment by austinsteffes** [comment]
@paouelle They are being used currently and there will be an effort to remove it all after continuous solr is integrated

File: platform/solr/solr-factory/src/main/java/org/codice/solr/factory/impl/ConfigurationFileProxy.java [hunk 1]
```diff
@@ -31,9 +31,11 @@
 import org.slf4j.LoggerFactory;
 
 /**
- * Abstraction layer for accessing files or directories on disk. Provides different implementations
- * depending on if the code is run within an OSGi container or not.
+ * @deprecated Intrigue and simple search replace this functionality. Abstraction layer for
```
Comment (pklinef) [inline platform/solr/solr-factory/src/main/java/org/codice/solr/factory/impl/ConfigurationFileProxy.java]: ✏️ We should not be referencing Catalog UI's like Intrigue at a platform level.

**Comment by austinsteffes** [comment]
build now

File: platform/solr/solr-factory/src/main/java/org/codice/solr/factory/impl/ConfigurationFileProxy.java (hunk 1 above)
Comment (austinsteffes) [reply platform/solr/solr-factory/src/main/java/org/codice/solr/factory/impl/ConfigurationFileProxy.java]: fixed to more generalized

**Comment by austinsteffes** [comment]
finalized javadocs. build now

**Pull request merged**